handsanitizer/DeclarationManager.cpp

```diff
@@ -0,0 +1,162 @@
+#include <vector>
+#include "DeclarationManager.h"
+#include "DeclarationManager.h"
+
+namespace handsanitizer {
+    const std::string CPP_ADDRESSING_DELIMITER = ".";
+    const std::string LVALUE_DELIMITER = "_";
+    const std::string POINTER_DENOTATION = "__p";
+
+    /*
+     * This generates a unique name that is also distinct compared to its argument!
+     * This is a required to make sure that there is no accidental way to generate correct code by incorrectly calling this
+     * As names generated by this should not conflict with argument  names, even though argument names can be passed in.
+     * If this function were not to guarantee uniqueness w.r.t it's arguments, then it would depend on the order which this function is called
+     * and implicit orderings are bad
+    */
+    std::string DeclarationManager::getUniqueTmpCPPVariableNameFor(std::vector<std::string> prefixes) {
+        auto candidate = joinStrings(prefixes, GENERATE_FORMAT_CPP_VARIABLE);
+
+        // This is explicitly not random
+        int counter = 0;
+        while (true) {
+            candidate = candidate + getRandomDummyVariableName();
+            counter++;
+
+            if (!this->isNameDefined(candidate)) {
+                definedNamesForFunctionBeingGenerated.push_back(candidate);
+                return candidate;
+            }
+        }
+    }
+
+    std::vector<std::string> TmpStringCandidates{
+            "foo",
+            "bar",
+            "baz",
+            "alice",
+            "bob",
+            "carlos",
+            "dan",
+            "erin",
+            "eve",
+            "judy",
+            "levi",
+            "micheal",
+            "panda", // hi my name is
+            "slim",
+            "shady",
+            "vera",
+            "victor",
+            "ted",
+    };
+
+    std::string DeclarationManager::getRandomDummyVariableName() {
```

Comment:
`TmpStringCandidates` lives longer than this function, so you can return `std::string&` here. 

You can also make this more defensive by marking both the function and the return as `const`:

```suggestion
    const std::string& DeclarationManager::getRandomDummyVariableName() const {
```